include/deal.II/grid/tria_accessor.h

```diff
@@ -5537,9 +5537,18 @@ TriaAccessor<structdim, dim, spacedim>::combined_face_orientation(
   if constexpr (structdim == 1)
     return ReferenceCell::default_combined_face_orientation();
   else if constexpr (structdim == 2)
-    return this->line_orientation(face) == true ?
-             ReferenceCell::default_combined_face_orientation() :
-             ReferenceCell::reversed_combined_line_orientation();
+    {
+      // if all elements are quads (or if we have a very special consistently
+      // oriented triangular mesh) then we do not store this array
+      if (this->tria->levels[this->present_level]
+            ->face_orientations.n_objects() == 0)
+        return ReferenceCell::default_combined_face_orientation();
+      else
+        return this->tria->levels[this->present_level]
+          ->face_orientations.get_orientation(
+            this->present_index * GeometryInfo<structdim>::faces_per_cell +
+            face);
```

Comment:
Like in the other patch, I assume this function is only called for hex or quad meshes? I'm asking because you are introducing a dependency on `GeometryInfo` where there was none before. (I recognize that you are only moving it.)

Reply:
Same comment applies here - the indexing requires knowing some undocumented implementation details.